internal/kafka/internal/services/kafka_tls_certificate_management/kafka_tls_certificate_management_service.go

```diff
@@ -0,0 +1,225 @@
+package kafka_tls_certificate_management
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+	"github.com/caddyserver/certmagic"
+	"github.com/libdns/route53"
+)
+
+// CertificateManagementOutput is the output indicating the certificates references
+type CertificateManagementOutput struct {
+	TLSCertRef string
+	TLSKeyRef  string
+}
+
+// GetCertificateRequest is the certificate request object.
+type GetCertificateRequest struct {
+	TLSCertRef string
+	TLSKeyRef  string
+}
+
+// Certificate is the content of the certificate
+type Certificate struct {
+	TLSCert string
+	TLSKey  string
+}
+
+// CertificateRevocationReason is the reason for the revocation of the certificates.
+// See https://www.rfc-editor.org/rfc/rfc5280#section-5.3.1 for the available reasons
+type CertificateRevocationReason int
+
+const (
+	Unspecified          CertificateRevocationReason = 0
+	KeyCompromise        CertificateRevocationReason = 1
+	CACompromise         CertificateRevocationReason = 2
+	AffiliationChanged   CertificateRevocationReason = 3
+	Superseded           CertificateRevocationReason = 4
+	CessationOfOperation CertificateRevocationReason = 5
+	CertificateHold      CertificateRevocationReason = 6
+	RemoveFromCRL        CertificateRevocationReason = 8
+	PrivilegeWithdrawn   CertificateRevocationReason = 9
+	AACompromise         CertificateRevocationReason = 10
+)
+
+func (reason CertificateRevocationReason) Int() int {
+	return int(reason)
+}
+
+//go:generate moq -out kafka_tls_certificate_management_service_moq.go . KafkaTLSCertificateManagementService
+type KafkaTLSCertificateManagementService interface {
+	// ManageCertificate manages wildcard tls certificate of a given domain automatically.
+	// If the certificate does not exists, it generates them
+	// If they exists, checks if they've expired and in this case it renews them before they do.
+	// It returns the keys referencing the wildcard certificates location in the Storage
+	ManageCertificate(ctx context.Context, domain string) (CertificateManagementOutput, error)
+
+	// GetCertificate returns the tls certificate given the request.
+	// The certificate are returned from the underlying certificate storage when certificate management is automatic
+	// otherwise, the certificate are returned from the manual tls configuration files
```

Comment:
```suggestion
	// The certificate is returned from the underlying certificate storage when certificate management is automatic
	// otherwise, the certificate is returned from the manual tls configuration files
```